pkg/controller/controller.go

```diff
@@ -1351,9 +1351,34 @@ func (s *Service) ListSnapshots(ctx context.Context, req *csi.ListSnapshotsReque
 	}, nil
 }
 
+// Returns the metro replication session ID for the volume.
+// If Metro is configured on the volume group that the volume is part of, then returns the ID of the volume group's session.
+func GetMetroSessionID(ctx context.Context, client gopowerstore.Client, volumeID string, volume gopowerstore.Volume, remoteArrayID string) (string, error) {
+	metroSessionID := ""
+	if remoteArrayID != "" {
```

Comment:
Invert the logic here to return an error if the remote array ID is empty.
as it is currently, it will just return an empty string and no error. This also allows you to bring all the functionality out of the `if` statement

Reply:
Updated!